ccl/storageccl: TestEncryptDecrypt failed

[(ccl/storageccl).TestEncryptDecrypt failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2615526&tab=buildLog) on [master@d86781c07065421f4a4d8bf5d988900ab07fdce5](https://github.com/cockroachdb/cockroach/commits/d86781c07065421f4a4d8bf5d988900ab07fdce5):

Fatal error:
```
panic: invalid argument to Int63n [recovered]
	panic: invalid argument to Int63n

```

Stack:
```
goroutine 134 [running]:
testing.tRunner.func1.1(0x3a7d700, 0x4cebc50)
	/usr/local/go/src/testing/testing.go:1072 +0x30d
testing.tRunner.func1(0xc000a0e180)
	/usr/local/go/src/testing/testing.go:1075 +0x41a
panic(0x3a7d700, 0x4cebc50)
	/usr/local/go/src/runtime/panic.go:969 +0x1b9
math/rand.(*Rand).Int63n(0xc0008ff9e0, 0x0, 0x0)
	/usr/local/go/src/math/rand/rand.go:111 +0x11d
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestEncryptDecrypt.func4.2(0xc000a0e180)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/encryption_test.go:194 +0x14b
testing.tRunner(0xc000a0e180, 0xc000a8ef60)
	/usr/local/go/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x2b3

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestEncryptDecrypt
--- FAIL: TestEncryptDecrypt (1.56s)
=== RUN   TestEncryptDecrypt/Random/ReadAt
        --- FAIL: TestEncryptDecrypt/Random/ReadAt (0.76s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestEncryptDecrypt PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestEncryptDecrypt.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestEncryptDecrypt failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2615527&tab=buildLog) on [master@d86781c07065421f4a4d8bf5d988900ab07fdce5](https://github.com/cockroachdb/cockroach/commits/d86781c07065421f4a4d8bf5d988900ab07fdce5):

Fatal error:
```
panic: invalid argument to Int63n [recovered]
	panic: invalid argument to Int63n

```

Stack:
```
goroutine 146 [running]:
testing.tRunner.func1.1(0x579b740, 0x7648610)
	/usr/local/go/src/testing/testing.go:1072 +0x46a
testing.tRunner.func1(0xc000102f00)
	/usr/local/go/src/testing/testing.go:1075 +0x636
panic(0x579b740, 0x7648610)
	/usr/local/go/src/runtime/panic.go:975 +0x47a
math/rand.(*Rand).Int63n(0xc000551860, 0x0, 0x0)
	/usr/local/go/src/math/rand/rand.go:111 +0x185
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestEncryptDecrypt.func4.2(0xc000102f00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/encryption_test.go:194 +0x16b
testing.tRunner(0xc000102f00, 0xc000d70a80)
	/usr/local/go/src/testing/testing.go:1123 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x5bc

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestEncryptDecrypt
--- FAIL: TestEncryptDecrypt (23.46s)
=== RUN   TestEncryptDecrypt/Random/ReadAt
        --- FAIL: TestEncryptDecrypt/Random/ReadAt (14.56s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-race -parallel=4

```
make stressrace TESTS=TestEncryptDecrypt PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestEncryptDecrypt.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestEncryptDecrypt failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2617702&tab=buildLog) on [master@c195665555d3a28a3e3857cb474f3ebe4e0c8d14](https://github.com/cockroachdb/cockroach/commits/c195665555d3a28a3e3857cb474f3ebe4e0c8d14):

Fatal error:
```
panic: invalid argument to Int63n [recovered]
	panic: invalid argument to Int63n

```

Stack:
```
goroutine 128 [running]:
testing.tRunner.func1.1(0x57a00c0, 0x764d770)
	/usr/local/go/src/testing/testing.go:1072 +0x46a
testing.tRunner.func1(0xc000434c00)
	/usr/local/go/src/testing/testing.go:1075 +0x636
panic(0x57a00c0, 0x764d770)
	/usr/local/go/src/runtime/panic.go:975 +0x47a
math/rand.(*Rand).Int63n(0xc0009d6390, 0x0, 0x0)
	/usr/local/go/src/math/rand/rand.go:111 +0x185
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestEncryptDecrypt.func4.2(0xc000434c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/encryption_test.go:194 +0x16b
testing.tRunner(0xc000434c00, 0xc000b88990)
	/usr/local/go/src/testing/testing.go:1123 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x5bc

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestEncryptDecrypt
--- FAIL: TestEncryptDecrypt (21.62s)
=== RUN   TestEncryptDecrypt/Random/ReadAt
        --- FAIL: TestEncryptDecrypt/Random/ReadAt (9.88s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestEncryptDecrypt PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestEncryptDecrypt.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/storageccl).TestEncryptDecrypt failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2618006&tab=buildLog) on [master@081f813ae4fb04653909d2eebd862e5965e21878](https://github.com/cockroachdb/cockroach/commits/081f813ae4fb04653909d2eebd862e5965e21878):

Fatal error:
```
panic: invalid argument to Int63n [recovered]
	panic: invalid argument to Int63n

```

Stack:
```
goroutine 144 [running]:
testing.tRunner.func1.1(0x57a5ae0, 0x7653a70)
	/usr/local/go/src/testing/testing.go:1072 +0x46a
testing.tRunner.func1(0xc000602600)
	/usr/local/go/src/testing/testing.go:1075 +0x636
panic(0x57a5ae0, 0x7653a70)
	/usr/local/go/src/runtime/panic.go:975 +0x47a
math/rand.(*Rand).Int63n(0xc0004df6b0, 0x0, 0x0)
	/usr/local/go/src/math/rand/rand.go:111 +0x185
github.com/cockroachdb/cockroach/pkg/ccl/storageccl.TestEncryptDecrypt.func4.2(0xc000602600)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/storageccl/encryption_test.go:195 +0x196
testing.tRunner(0xc000602600, 0xc000d933e0)
	/usr/local/go/src/testing/testing.go:1123 +0x203
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x5bc

```

<details><summary>Log preceding fatal error</summary><p>

```
=== RUN   TestEncryptDecrypt
--- FAIL: TestEncryptDecrypt (32.46s)
=== RUN   TestEncryptDecrypt/Random/ReadAt
        --- FAIL: TestEncryptDecrypt/Random/ReadAt (20.56s)

```

</p></details>

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestEncryptDecrypt PKG=./pkg/ccl/storageccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestEncryptDecrypt.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/ccl/storageccl/encryption_test.go
+++ b/pkg/ccl/storageccl/encryption_test.go
@@ -191,8 +191,8 @@ func TestEncryptDecrypt(t *testing.T) {
 					r, err := decryptingReader(bytes.NewReader(ciphertext), key)
 					require.NoError(t, err)
 					for k := 0; k < reads; k++ {
-						start := rng.Int63n(int64(float64(len(plaintext)) * 1.1))
-						expected := make([]byte, rng.Int63n(int64(len(plaintext))/2))
+						start := rng.Int63n(int64(float64(len(plaintext)+1) * 1.1))
+						expected := make([]byte, rng.Int63n(int64(len(plaintext)/2+1)))
 						got := make([]byte, len(expected))
 						expectedN, expectedErr := plainReader.ReadAt(expected, start)
 						gotN, gotErr := r.(io.ReaderAt).ReadAt(got, start)

